File: src/pers/gargantua/mine_sweep/Maps.java
package pers.gargantua.mine_sweep;

import java.util.ArrayList;
import java.util.Random;

public final class Maps extends ArrayList<ArrayList<Ground>> {
    
    private static Maps maps = null;
    
    public Config config;
    
    private Maps() {
    }
    
    public static Maps getMaps() {
        if (maps == null)
            maps = new Maps();
        return maps;
    }
    
    public void init() {
        clear();
        for (int i = 0; i < config.r; i++) {
            ArrayList<Ground> list = new ArrayList<>();
            for (int j = 0; j < config.c; j++) {
                list.add(new Ground(i, j));
            }
            add(list);
        }
        
        Random random = new Random();
        for (int i = 0; i < config.m; i++) {
            while (true) {
                Ground ground = get(random.nextInt(config.r), random.nextInt(config.c));
                if (!ground.isMine) {
                    ground.isMine = true;
                    break;
                }
            }
        }
        
        for (int i = 0; i < config.r; i++) {
            for (int j = 0; j < config.c; j++) {
                if (this.get(i, j).isMine) continue;
                int quantityOfMine = 0;
                for (int r = i - 1; r <= i + 1; r++) {
                    for (int c = j - 1; c <= j + 1; c++) {
                        if ((i != r || j != c) && r >= 0 && r < config.r && c >= 0 && c < config.c &&
                                get(r, c).isMine){
                            quantityOfMine++;
                        }
                    }
                }
                get(i, j).around = quantityOfMine;
            }
        }
    }
    
    public Ground get(int r, int c) {
        return this.get(r).get(c);
    }
}

File: src/pers/gargantua/mine_sweep/Maps.kt
package pers.gargantua.mine_sweep

import java.util.*
import kotlin.collections.ArrayList

/**
 * @author Gargantua丶
 **/
object Maps : ArrayList<ArrayList<Ground>>() {

    lateinit var config: Config

    fun initialize() {
        clear()
        for (i in 0 until config.r) {
            add(ArrayList<Ground>().apply {
                for (j in 0 until config.c)
                    this.add(Ground(i, j))
            })
        }

        val random = Random()
        for (i in 0 until config.m) {
            while (true) {
                val ground = this[random.nextInt(config.r)][random.nextInt(config.c)]
                if (!ground.isMine) {
                    ground.isMine = true
                    break
                }
            }
        }

        for (i in 0 until config.r) {
            for (j in 0 until config.c) {
                if (this[i][j].isMine) continue
                var quantityOfMine = 0
                for (r in i - 1..i + 1) {
                    for (c in j - 1..j + 1) {
                        if ((i != r || j != c) && r in 0 until config.r && c in 0 until config.c &&
                            this[r][c].isMine) {
                            quantityOfMine++
                        }
                    }
                }
                this[i][j].around = quantityOfMine
            }
        }

        for (grounds in this) {
            for (ground in grounds) print("${ground.around} ")
            println()
        }
    }
}